### ISSUE TITLE
make command failing for htop 2.1.0 

### ISSUE THREAD

**Issue by anujajakhade** [hishamhm/htop#744]
I am building htop 2.1.0 from source on _SUSE Linux Enterprise Server 12 SP3  (x86_64)_.
./configure command is running successfully. However make command fails with the error.

 ```
 htop-AvailableMetersPanel.o htop-CategoriesPanel.o htop-CheckItem.o htop-ClockMeter.o htop-ColorsPanel.o htop-ColumnsPanel.o htop-CPUMeter.o htop-CRT.o htop-MainPanel.o htop-DisplayOptionsPanel.o htop-FunctionBar.o htop-Hashtable.o htop-Header.o htop-htop.o htop-ListItem.o htop-LoadAverageMeter.o htop-MemoryMeter.o htop-Meter.o htop-MetersPanel.o htop-Object.o htop-Panel.o htop-BatteryMeter.o htop-Process.o htop-ProcessList.o htop-RichString.o htop-ScreenManager.o htop-Settings.o htop-SignalsPanel.o htop-StringUtils.o htop-SwapMeter.o htop-TasksMeter.o htop-UptimeMeter.o htop-TraceScreen.o htop-UsersTable.o htop-Vector.o htop-AvailableColumnsPanel.o htop-AffinityPanel.o htop-HostnameMeter.o htop-OpenFilesScreen.o htop-Affinity.o htop-IncSet.o htop-Action.o htop-EnvScreen.o htop-InfoScreen.o htop-XAlloc.o linux/htop-Platform.o linux/htop-IOPriorityPanel.o linux/htop-IOPriority.o linux/htop-LinuxProcess.o linux/htop-LinuxProcessList.o linux/htop-LinuxCRT.o linux/htop-Battery.o  -L/usr/lib64/ncurses6 -lncursesw -ltinfo -lm
htop-htop.o: In function `main':
/htop-2.1.0/htop.c:233: undefined reference to `stdscr'
/htop-2.1.0/htop.c:234: undefined reference to `LINES'
/htop-2.1.0/htop.c:234: undefined reference to `stdscr'
/htop-2.1.0/htop.c:234: undefined reference to `COLS'
/htop-2.1.0/htop.c:234: undefined reference to `stdscr'
/htop-2.1.0/htop.c:235: undefined reference to `stdscr'
/htop-2.1.0/htop.c:236: undefined reference to `stdscr'
htop-CRT.o: In function `CRT_readKey':
/htop-2.1.0/CRT.c:698: undefined reference to `stdscr'
/htop-2.1.0/CRT.c:699: undefined reference to `stdscr'
htop-CRT.o:/htop-2.1.0/CRT.c:707: more undefined references to `stdscr' follow
htop-FunctionBar.o: In function `FunctionBar_drawAttr':
/htop-2.1.0/FunctionBar.c:105: undefined reference to `LINES'
/htop-2.1.0/FunctionBar.c:105: undefined reference to `COLS'
/htop-2.1.0/FunctionBar.c:105: undefined reference to `stdscr'
/htop-2.1.0/FunctionBar.c:108: undefined reference to `stdscr'
/htop-2.1.0/FunctionBar.c:109: undefined reference to `LINES'
/htop-2.1.0/FunctionBar.c:109: undefined reference to `stdscr'
/htop-2.1.0/FunctionBar.c:111: undefined reference to `stdscr'
/htop-2.1.0/FunctionBar.c:112: undefined reference to `LINES'
/htop-2.1.0/FunctionBar.c:112: undefined reference to `stdscr'
/htop-2.1.0/FunctionBar.c:116: undefined reference to `stdscr'
/htop-2.1.0/FunctionBar.c:117: undefined reference to `LINES'
/htop-2.1.0/FunctionBar.c:117: undefined reference to `stdscr'
/htop-2.1.0/FunctionBar.c:123: undefined reference to `stdscr'
htop-Header.o: In function `Header_draw':
/htop-2.1.0/Header.c:183: undefined reference to `stdscr'
```

Investigating further, I observed that the Makefile has

> LIBS = -L/usr/lib64/ncurses6 -lncursesw -ltinfo -lm

If I change the above line to
>LIBS =  -lncursesw -ltinfo -lm

Am able to successfully run the make and make install command.

Is there any other way to resolve this issue?

**Comment by hishamhm** [comment]
That's odd: this `LIBS` line should have been detected/generated by the appropriate `ncurses*-config` script in your system. Can you try again with the latest `master`?

**Comment by anujajakhade** [comment]
@hishamhm , Thank you for your reply.
I could successfully build htop on master. Is there any way to fix the issue in v2.1.0.

**Comment by jschleus** [comment]
As already reported within a personal mail to  hisham a month ago I have had the same problem on a related openSUSE Leap 42.3 system and I can confirm that the above change of  anujajakhade solved also my linking problem.

Out of curiosity I am searched for the probably involved `ncurses*-config` script and found `/usr/bin/ncursesw6-config` with the lines
```
# compile/link
--cflags)
            echo -I/usr/include/ncurses6/ncursesw -I/usr/include/ncurses6
            ;;
--libs)
            echo -L/usr/lib64/ncurses6 -lncursesw -ltinfo
            ;;
```
So I used the originally created Makefile and changed testwise the line
```
CFLAGS = -g -O2
```
into
```
CFLAGS = -I/usr/include/ncurses6 -g -O2
```
and it works also (`-I/usr/include/ncurses6/ncursesw` doesn't seem necessary).

And yes, the latest `master` links also correctly (probably since in `configure.ac` now something like `htop_config_script_cflags=$([$4] --cflags` is used, but I'm a layman in this area)..

**Comment by gavinhungry** [comment]
Same issue here on SLES 11 SP4, `master` branch (as of [`697f5bb`](https://github.com/hishamhm/htop/tree/697f5bb9c14a89b2e04519c956fc7e455c9b1cc0)) worked for me.

**Comment by Nayana-ibm** [comment]
@hishamhm  Any plans to backport fix to v2.1.0? Or can you point me exact commit which fixed the issue?

**Comment by hishamhm** [comment]
@Nayana-ibm 2.1.1 is coming soon!

**Comment by Nayana-ibm** [comment]
@hishamhm Please let me know about release/tag which resolves above issue

**Comment by anujajakhade** [comment]
Looks like fixed in v2.2.0.
